core/src/main/java/org/apache/iceberg/TableMetadataParser.java

```diff
@@ -299,7 +299,7 @@ public static TableMetadata fromJson(String metadataLocation, String json) {
     return JsonUtil.parse(json, node -> TableMetadataParser.fromJson(metadataLocation, node));
   }
 
-  static TableMetadata fromJson(InputFile file, JsonNode node) {
+  public static TableMetadata fromJson(InputFile file, JsonNode node) {
```

Comment:
@Fokko, @snazy, I think we need to revert this. We cannot leak Jackson classes publicly through an Iceberg API, even in core.

Reply:
I talked to Fokko about this and I think it should be okay. I didn't realize we were already doing this elsewhere. Sorry for the confusion!